### ISSUE TITLE
Do I need some initialization ?

### ISSUE THREAD

**Issue by j-fu** [tro3/ThreadPools.jl#7]
Hi, I am trying out the package, and  I have  a problem running my loops with `@qthreads`.
Do I need to peform some initialization before using the macros ?

Here is the MWE:
````
using ThreadPools
using BenchmarkTools

function qrun(N)
    a=rand(N)
    b=rand(N)
    ThreadPools.@qthreads for i=1:N
        a[i]*=b[i]
    end
end

qrun(10)
@time qrun(100000)
````
It gives:
````
ERROR: LoadError: InvalidStateException("Channel is closed.", :closed)
Stacktrace:
 [1] check_channel_state at ./channels.jl:167 [inlined]
 [2] take_buffered(::Channel{Task}) at ./channels.jl:399
 [3] take! at ./channels.jl:394 [inlined]
 [4] take! at /home/fuhrmann/.julia/packages/ThreadPools/3VA3A/src/qpool.jl:93 [inlined]
 [5] pmap(::ThreadPools.QueuePool, ::var"#4#6"{Array{Float64,1},Array{Float64,1}}, ::UnitRange{Int64}) at /home/fuhrmann/.julia/packages/ThreadPools/3VA3A/src/qpool.jl:197
 [6] pforeach at /home/fuhrmann/.julia/packages/ThreadPools/3VA3A/src/interface.jl:33 [inlined]
 [7] pforeach at /home/fuhrmann/.julia/packages/ThreadPools/3VA3A/src/interface.jl:37 [inlined]
 [8] #3 at /home/fuhrmann/.julia/packages/ThreadPools/3VA3A/src/macros.jl:18 [inlined]
 [9] pwith at /home/fuhrmann/.julia/packages/ThreadPools/3VA3A/src/interface.jl:101 [inlined]
 [10] macro expansion at /home/fuhrmann/.julia/packages/ThreadPools/3VA3A/src/macros.jl:17 [inlined]
 [11] qrun(::Int64) at /home/fuhrmann/Wias/work/julia/julia-tests/parallel/tmwe.jl:7
 [12] top-level scope at util.jl:155
 [13] include at ./boot.jl:328 [inlined]
 [14] include_relative(::Module, ::String) at ./loading.jl:1105
 [15] include(::Module, ::String) at ./Base.jl:31
 [16] exec_options(::Base.JLOptions) at ./client.jl:287
 [17] _start() at ./client.jl:460
in expression starting at /home/fuhrmann/Wias/work/julia/julia-tests/parallel/tmwe.jl:13
````

**Comment by tro3** [comment]
!! This is a bug.  It looks like the macro is leaving the underlying pool in the global space instead of the function space, so the second time you run the function, it tries to use the same one and dies.  In the meantime, this is what the macro is supposed to expand to, and it seems to work:

```
function qrun(N)
  a=rand(N)
  b=rand(N)
  pwith(ThreadPools.QueuePool()) do pool
    pforeach(pool, 1:N) do i
      a[i]*=b[i]
    end
  end
end
```

**Comment by tro3** [comment]
Okay, I see what's going on.  I'll rev this to 1.0.1 tonight.  I also have to deprecae `pmap` et al for `tmap`.

**Comment by tro3** [comment]
Closed by #9

**Comment by tro3** [comment]
@j-fu - thanks for the feedback.  This issue is fixed in 1.0.1, just registered.
